src/Rings/MPolyQuo.jl

```diff
@@ -1550,50 +1550,38 @@ return an array containing a minimal set of generators of `a`.
 ```jldoctest
 julia> R, (x, y, z) = graded_polynomial_ring(QQ, ["x", "y", "z"]);
 
-julia> V = [x, z^2, x^3+y^3, y^4, y*z^5];
-
-julia> I = ideal(R, V)
-ideal(x, z^2, x^3 + y^3, y^4, y*z^5)
-
 julia> A, p = quo(R, ideal(R, [x-y]));
 
-julia> a = ideal(A, [p(x) for x in V]);
+julia> V = [x, z^2, x^3+y^3, y^4, y*z^5];
+
+julia> a = ideal(A, V);
 
 julia> minimal_generating_set(a)
 2-element Vector{MPolyQuoRingElem{MPolyDecRingElem{QQFieldElem, QQMPolyRingElem}}}:
- x
+ y
  z^2
 ```
 """
-function minimal_generating_set(I::MPolyQuoIdeal{<:MPolyDecRingElem}; ordering::MonomialOrdering = default_ordering(base_ring(base_ring(I))))
+function minimal_generating_set(I::MPolyQuoIdeal{<:MPolyDecRingElem})
   # This only works / makes sense for homogeneous ideals. So far ideals in an
   # MPolyDecRing are forced to be homogeneous though.
-
   Q = base_ring(I)
 
   @assert is_graded(Q)
 
   @req coefficient_ring(Q) isa AbstractAlgebra.Field "The coefficient ring must be a field"
 
-  QS = singular_poly_ring(Q)
-  singular_assure(I)
-
-  IS = I.gens.S
-  GC.@preserve IS QS begin
-    ptr = Singular.libSingular.idMinBase(IS.ptr, QS.ptr)
-    gensS = gens(typeof(IS)(QS, ptr))
-  end
-
-  i = 1
-  while i <= length(gensS)
-    if iszero(gensS[i])
-      deleteat!(gensS, i)
```

Comment:
Without this, the following happens:
```
julia> R, (x, y) = grade(polynomial_ring(QQ, [ "x", "y"])[1], [ 1, 2 ])
(Multivariate Polynomial Ring in x, y over Rational Field graded by 
  x -> [1]
  y -> [2], MPolyDecRingElem{QQFieldElem, QQMPolyRingElem}[x, y])

julia> I = ideal(R, [ x*y ])
ideal(x*y)

julia> Q, _ = quo(R, I)
(Quotient of Multivariate Polynomial Ring in x, y over Rational Field graded by 
  x -> [1]
  y -> [2] by ideal(x*y), Map from
R to Q defined by a julia-function with inverse)

julia> J = ideal(Q, [ x*y ])
ideal(x*y)

julia> minimal_generating_set(J)
1-element Vector{MPolyQuoRingElem{MPolyDecRingElem{QQFieldElem, QQMPolyRingElem}}}:
 0
```
So an ideal that is trivial in the quotient ring gets minimal generating set `[0]`. This is why the tests in the `LinearQuotient`-code fail. Is this behaviour of `minimal_generating_set` intended?

Reply:
Thanks for the hint! I just pushed a change to remove zero from the list of generators.

Reply:
Thanks!

Reply:
So what's the return value for the zero ideal now?

Reply:
An empty list, this should be the same behaviour as before.

Reply:
Both Macaulay2 and Singular return 0. There must be reasons for this: @hannes14

Reply:
That is a convention: Singular generates the zero ideal by a single 0.

Reply:
I can also make a small fix to make my code (and the tests) work with `[0]`, if that is the convention we follow.

Reply:
I talked to @fieker and we agree in that we think there is no "best" solution for this. So let us pick one solution: I suggest to go along with Macaulay2 and Singular: We return the vector consistsing of the zero element of the given ring.

Reply:
Is it allowed to ask why? This is not a minimal generating set and contradicts the docstring. This is confusing from a user point of view.

Reply:
(A doctest with a zero ideal might be useful, too)

Reply:
Just for entertainment: there is a test (not a doctest) testing this:
```
R, (x, y) = grade(polynomial_ring(QQ, [ "x", "y"])[1], [ 1, 2 ])
@test minimal_generating_set(ideal(R, [ R() ])) == elem_type(R)[]
```
(see `test/Rings/mpoly-graded-test.jl` line 212). That test works also with the earlier version of this pull request... Apparently Singular does not always return a non-empty list?

Reply:
>Whatever we do, I think it'd be worthwhile to point this out in the manual, as either behavior will throw off some people: returning [0] is not minimal, so must be mentioned; returning an empty list is just something people forget can happen, so it's also helpful to mention.

See https://github.com/oscar-system/Oscar.jl/pull/2291/commits/2f89770702d0c7e2f13f66565300dce9c131ba0f for a corresponding change to the docs, let me know if I should also add it explicitly as a doc test.

Reply:
@RafaelDavidMohr Thank you for changing the docs, this looks fine to me. Please add it also to the doc tests in both cases. @fingolfin Otherwise, from my point of view, this is ready to me merged.